src/cufinufft.cu

```diff
@@ -315,23 +381,196 @@ Notes: the type FLT means either single or double, matching the
         cudaGetDevice(& orig_gpu_device_id);
         cudaSetDevice(d_plan->opts.gpu_device_id);
 
+	int ier;
+
+	int dim = d_plan->dim;
+
+	d_plan->M = M;
+
+	if(d_plan->type == 3) {
+		d_plan->N = N;
+
+		// pick x, s intervals & shifts & # fine grid pts (nf) in each dim...
+		FLT S1,S2,S3;       // get half-width X, center C, which contains {x_j}...
+		arraywidcen_gpu(M, d_kx, &(d_plan->t3P.X1), &(d_plan->t3P.C1));
+		arraywidcen_gpu(N, d_s, &S1, &(d_plan->t3P.D1));      // same D, S, but for {s_k}
+		set_nhg_type3(S1, d_plan->t3P.X1, d_plan->opts, d_plan->spopts,
+			&(d_plan->nf1), &(d_plan->t3P.h1), &(d_plan->t3P.gam1));  // applies twist i)
+
+		d_plan->t3P.C2 = 0.0;        // their defaults if dim 2 unused, etc
+		d_plan->t3P.D2 = 0.0;
+		if (d_plan->dim>1) {
```

Comment:
So it looks like a lot of this code would support dimensions lower than 3. What's missing to get that to work?

Reply:
I've written everything with all three dimensions in mind. Supporting 1d and 2d only requires some additional work, replicating the execution in 3d. The `setpts` function should fully support all dimensions already.

Reply:
Ok cool.